man/systemd-ssh-generator.xml

```diff
@@ -0,0 +1,141 @@
+    exists, and otherwise generate a suitable service template file.</para>
+
+    <para><filename>systemd-ssh-generator</filename> implements
+    <citerefentry><refentrytitle>systemd.generator</refentrytitle><manvolnum>7</manvolnum></citerefentry>.</para>
+  </refsect1>
+
+  <refsect1>
+    <title>Kernel Command Line</title>
+
+    <para><filename>systemd-ssh-generator</filename> understands the following
+    <citerefentry><refentrytitle>kernel-command-line</refentrytitle><manvolnum>7</manvolnum></citerefentry>
+    parameters:</para>
+
+    <variablelist class='kernel-commandline-options'>
+      <varlistentry>
+        <term><varname>systemd.ssh_auto=</varname></term>
+
+        <listitem><para>This options take an optional boolean argument, and default to yes. If on the
+        automatic binding to the <constant>AF_VSOCK</constant> and <constant>AF_UNIX</constant> sockets
+        listed above is done. If off, this is not done, except for those explicitly requested via
+        <varname>systemd.ssh_listen=</varname> on the kernel command line or via the
+        <varname>ssh.listen</varname> system credential.</para>
+
+        <xi:include href="version-info.xml" xpointer="v256"/></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>systemd.ssh_listen=</varname></term>
+
+        <listitem><para>This option configures an additional socket to bind SSH to. May be used multiple
+        times to bind multiple sockets. The syntax should follow the one of
+        <varname>ListenStream=</varname>, see
+        <citerefentry><refentrytitle>systemd.socket</refentrytitle><manvolnum>5</manvolnum></citerefentry>
+        for details. This functionality supports all socket families systemd supports, including
+        <constant>AF_INET</constant> and <constant>AF_INET6</constant>.</para>
+
+        <xi:include href="version-info.xml" xpointer="v256"/></listitem>
+      </varlistentry>
+    </variablelist>
+  </refsect1>
+
+  <refsect1>
+    <title>Credentials</title>
+
+    <para><command>systemd-ssh-generator</command> supports the system credentials logic. The following
+    credentials are used when passed in:</para>
+
+    <variablelist class='system-credentials'>
+      <varlistentry>
+        <term><varname>ssh.listen</varname></term>
+
+        <listitem><para>This credential should be a text file, with each line referencing one additional
+        socket to bind SSH to. The syntax should follow the one of <varname>ListenStream=</varname>, see
+        <citerefentry><refentrytitle>systemd.socket</refentrytitle><manvolnum>5</manvolnum></citerefentry>
+        for details. This functionality supports all socket families systemd supports, including
+        <constant>AF_INET</constant> and <constant>AF_INET6</constant>.</para>
+
+        <xi:include href="version-info.xml" xpointer="v256"/></listitem>
+      </varlistentry>
+    </variablelist>
```

Comment:
This seems messy, why not make it take a credential glob?

Reply:
Also, why not allow configuring `ssh_auto` via a credential? And why do kernel command line at all for these and not just credentials?

Reply:
The only reason for systemd.ssh_auto= to exist is to appease people who hate everything, and those kind of people probably think system credentials are stupid too, so why offer it at all.

also, i am not sure I think arbitrary single-effect config knobs are really what credentials are supposed to be used for. for me credentials are more for payload than for config, but the line is blurry. i.e. i see creds as useful not singular, isolated knobs, but more comprehensive, larger config blobs, that combine a bunch of things into one.

at least that's my thinking. I mean, i am sure people will see this differently than I will, people will bend them to work for anything they want. But I think we should probably have a rough idea what we want them to be used for and for what kernel command line options are better. And this doesn't have to be a strict one or the other thing, but is a blurry line I am sure.

Reply:
Or to say this differently: let's start with the obvious uses. We can always extend things later.

While the case for ssh_listen is clear to me that we want both, but for ssh_auto i am not sure